libr/search/aes-find.c

```diff
@@ -30,20 +30,56 @@ const ut8 table_sbox[256] = {
 	0x8c, 0xa1, 0x89, 0x0d, 0xbf, 0xe6, 0x42, 0x68, 0x41, 0x99, 0x2d, 0x0f, 0xb0, 0x54, 0xbb, 0x16
 };
 
-static int aes_key_test(const unsigned char *buf) {
-	return (buf[28]==table_sbox[buf[15]] \
-	    &&  buf[29]==table_sbox[buf[12]] \
-	    &&  buf[30]==table_sbox[buf[13]] \
-	    &&  buf[31]==(table_sbox[buf[14]]^1))?1:0;
+static int aes256_key_test(const unsigned char *buf) {
+	int aes256;
+
+	aes256 = buf[32] == (buf[0] ^ table_sbox[buf[29]] ^ 1) \
+		&& buf[33] == (buf[1] ^ table_sbox[buf[30]]) \
+		&& buf[34] == (buf[2] ^ table_sbox[buf[31]]) \
+		&& buf[35] == (buf[3] ^ table_sbox[buf[28]])?1:0;
+
+	return (aes256);
+}
+
+static int aes192_key_test(const unsigned char *buf) {
+	int aes192;
+
+	aes192 = buf[24] == (buf[0] ^ table_sbox[buf[21]] ^ 1) \
+		&& buf[25] == (buf[1] ^ table_sbox[buf[22]]) \
+		&& buf[26] == (buf[2] ^ table_sbox[buf[23]]) \
+		&& buf[27] == (buf[3] ^ table_sbox[buf[20]])?1:0;
+
+	return (aes192);
+}
+
+static int aes128_key_test(const unsigned char *buf) {
+	int aes128;
```

Comment:
just reutnr this... no need for a local variable

also, i think you want to return bool in here and drop the ?1:0;